src/Manipulator/ObjectManipulator.php

```diff
@@ -0,0 +1,145 @@
+{
+    /**
+     * Add the instance to the object provided.
+     *
+     * @phpstan-template T of object
+     * @phpstan-param T $instance
+     * @phpstan-return T
+     */
+    public static function addInstance(
+        object $object,
+        object $instance,
+        FieldDescriptionInterface $parentFieldDescription
+    ): object {
+        $associationMapping = $parentFieldDescription->getAssociationMapping();
+        $parentAssociationMappings = $parentFieldDescription->getParentAssociationMappings();
+
+        foreach ($parentAssociationMappings as $parentAssociationMapping) {
+            $object = self::callGetter($object, $parentAssociationMapping['fieldName']);
+        }
+
+        return self::callAdder($object, $instance, $associationMapping['fieldName']);
+    }
+
+    /**
+     * Set the object to the instance provided.
+     *
+     * @phpstan-template T of object
+     * @phpstan-param T $instance
+     * @phpstan-return T
+     */
+    public static function setObject(
+        object $instance,
+        object $object,
+        FieldDescriptionInterface $parentFieldDescription
+    ): object {
+        $associationMapping = $parentFieldDescription->getAssociationMapping();
+        $parentAssociationMappings = $parentFieldDescription->getParentAssociationMappings();
+
+        foreach ($parentAssociationMappings as $parentAssociationMapping) {
+            $object = self::callGetter($object, $parentAssociationMapping['fieldName']);
+        }
+
+        return self::callSetter($instance, $object, $associationMapping['mappedBy']);
+    }
+
+    /**
+     * Call $object->getXXX().
+     */
+    private static function callGetter(object $object, string $fieldName): object
+    {
+        $inflector = InflectorFactory::create()->build();
+        $method = sprintf('get%s', $inflector->classify($fieldName));
+
+        if (!(\is_callable([$object, $method]) && method_exists($object, $method))) {
+            /*
+             * NEXT_MAJOR: Use BadMethodCallException instead
+             */
+            throw new \RuntimeException(
+                sprintf('Method %s::%s() does not exist.', ClassUtils::getClass($object), $method)
+            );
+        }
+
+        return $object->$method();
+    }
+
+    /**
+     * Call $instance->setXXX($object).
+     *
+     * @phpstan-template T of object
+     * @phpstan-param T $instance
+     * @phpstan-return T
+     */
+    private static function callSetter(object $instance, object $object, string $mappedBy): object
+    {
+        $inflector = InflectorFactory::create()->build();
+        $method = sprintf('set%s', $inflector->classify($mappedBy));
+
+        if (!(\is_callable([$instance, $method]) && method_exists($instance, $method))) {
+            /*
+             * NEXT_MAJOR: Use BadMethodCallException instead
+             */
+            throw new \RuntimeException(
+                sprintf('Method %s::%s() does not exist.', ClassUtils::getClass($instance), $method)
```

Comment:
@VincentLanglet Why did you stop using a Symfony PropertyAccess? I got this exception because my DTO has no setters and getters, but only public properties.

Reply:
We can still add a check for a public property. Do you want to create the PR ?

Reply:
> But I didn't change the strategy in this PR, it was already like this in `AdminHelper:: addNewInstance`.

So i messed up something.

Reply:
> > But I didn't change the strategy in this PR, it was already like this in `AdminHelper:: addNewInstance`.
> 
> So i messed up something.

I did review again, and I changed something

Previously, only `getter` and `adder` was used.
Now, `getter` and `setter` or `adder` are used.

Instead of `$parent->addChild()`, I sometimes call `$child->setParent()`.